system-and-security-architecture.rst: update based on recent swupd integration

pohly opened the pull request:
This is the revision of the document what we expect to release in the
next Ostro OS milestone.

Ismo and I also looked at
https://github.com/pohly/meta-ostro/blob/architecture-doc-cleanup/doc/architecture/security-threat-analysis.rst
and came to the conclusion that it is okay as it stands, i.e. no further changes are planned.

Signed-off-by: Patrick Ohly <patrick.ohly@intel.com>

cschaufler commented:
I don't see any significant issues with the document as changed.

mythi commented:
@pohly thanks!

dbkinder commented on doc/architecture/system-and-security-architecture.rst:
```diff
@@ -235,7 +235,7 @@ User, Group and Privilege Management
 User and group management files (like ``/etc/passwd``) are
 read-only. That means that the core system can only have static system
-users. It is not possible to set a root password.
+users. It is not supported to set a root password.
```
Change to:  "Setting a root password is not supported."

dbkinder commented on doc/architecture/system-and-security-architecture.rst:
```diff
@@ -435,7 +435,8 @@ use them only in secure environments.
 The Ostro Project provides different pre-compiled images. All of them
 are compiled as development images. Currently there are no pre-compiled
-production images.
+production images. Those pre-compiled image get security updates
```
Let's merge these short sentences to: 
The Ostro Project provides pre-compiled development images that can receive security updates using the software update mechanism.  No pre-compiled production images are provided.

dbkinder commented:
[skip ci]

ereshetova commented:
I agree with David suggestions, otherwise changes look good to me. One thing I noticed is that references don't work in doc. For example: " see :ref:`software-update`" and others. This is smth I guess that needs fixing.

dbkinder commented:
@ereshetova the :ref:`name` links don't work in the GitHub previewer, but will work once the content is rendered for the website

ereshetova commented:
Ok, great, then +1 on the changes from me.

mythi commented:
@pohly can you update this based on David's feedback and then let's merge.

pohly commented:
On Thu, 2016-04-14 at 00:42 -0700, Mikko Ylinen wrote:
> @pohly can you update this based on David's feedback and then let's
> merge.

Updated, rebased and force-pushed. Feel free to merge.

okartau commented:
retest this please
